### PR TITLE
BUG: compute.use_numexpr option not respected

### LINKED ISSUE #32556: Dataframe.query uses numexpr even when 'compute.use_numexpr' is False
#### Code Sample, a copy-pastable example if possible

```python
# Your code here
import pandas as pd
import io

content = io.StringIO('''
time,val
212.23, 32
''')

pd.set_option('compute.use_numexpr', False)
print("use numexpr? %d" % pd.get_option('compute.use_numexpr'))

df = pd.read_csv(
    content,
    sep=',',
    usecols=['time', 'val'],
    dtype= {'val': 'Int64'},
    # parse_dates=date_cols,
)
q = "val == 32"
# passing `engine="python"` fixes it
df_dest = df.query(q, )

```
throws with pandas 1.0
```
Traceback (most recent call last):
  File "test.py", line 21, in <module>
    df_dest = df.query(q, )
  File "/nix/store/0yyjswv58jk0fv1miyklmppvpd0p45sy-python3.7-pandas-1.0.1/lib/python3.7/sit
e-packages/pandas/core/frame.py", line 3231, in query
    res = self.eval(expr, **kwargs)
  File "/nix/store/0yyjswv58jk0fv1miyklmppvpd0p45sy-python3.7-pandas-1.0.1/lib/python3.7/sit
e-packages/pandas/core/frame.py", line 3346, in eval
    return _eval(expr, inplace=inplace, **kwargs)
  File "/nix/store/0yyjswv58jk0fv1miyklmppvpd0p45sy-python3.7-pandas-1.0.1/lib/python3.7/sit
e-packages/pandas/core/computation/eval.py", line 337, in eval
    ret = eng_inst.evaluate()
  File "/nix/store/0yyjswv58jk0fv1miyklmppvpd0p45sy-python3.7-pandas-1.0.1/lib/python3.7/sit
e-packages/pandas/core/computation/engines.py", line 73, in evaluate
    res = self._evaluate()
  File "/nix/store/0yyjswv58jk0fv1miyklmppvpd0p45sy-python3.7-pandas-1.0.1/lib/python3.7/sit
e-packages/pandas/core/computation/engines.py", line 114, in _evaluate
    return ne.evaluate(s, local_dict=scope)
  File "/nix/store/k6wr7w16q0aq9mfkf0w15wxlhx76sxqc-python3.7-numexpr-2.7.1/lib/python3.7/si
te-packages/numexpr/necompiler.py", line 822, in evaluate
    zip(names, arguments)]
  File "/nix/store/k6wr7w16q0aq9mfkf0w15wxlhx76sxqc-python3.7-numexpr-2.7.1/lib/python3.7/si
te-packages/numexpr/necompiler.py", line 821, in <listcomp>
    signature = [(name, getType(arg)) for (name, arg) in
  File "/nix/store/k6wr7w16q0aq9mfkf0w15wxlhx76sxqc-python3.7-numexpr-2.7.1/lib/python3.7/si
te-packages/numexpr/necompiler.py", line 703, in getType
    raise ValueError("unknown type %s" % a.dtype.name)
ValueError: unknown type object
```

#### Problem description

Dataframe.query fails when using UInt* dtypes see https://github.com/pandas-dev/pandas/issues/25369#issuecomment-594905994.
My first idea to solve this was to disable numexpr but this isn't enough, one has to pass  `engine="python"` to forcefully disable numexpr when calling `query`

### PR DESCRIPTION
**Pull request by saehuihwang**
- [x] closes #32556
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/computation/test_eval.py [hunk 1]
```diff
@@ -1865,6 +1865,16 @@ def test_invalid_engine():
         pd.eval("x + y", local_dict={"x": 1, "y": 2}, engine="asdf")
 
 
+@td.skip_if_no_ne
```
Comment (jreback) [inline pandas/tests/computation/test_eval.py:1868]: can you parameterize and text for both True/False here.

File: pandas/tests/computation/test_eval.py [hunk 2]
```diff
@@ -1865,6 +1865,16 @@ def test_invalid_engine():
...
+def test_numexpr_option_respected():
+    # GH 32556
+    from pandas.core.computation.eval import _check_engine
+
+    with pd.option_context("compute.use_numexpr", False):
```
Comment (jreback) [inline pandas/tests/computation/test_eval.py:1873]: can you add a test similar to https://github.com/pandas-dev/pandas/issues/32556#issuecomment-623364261

that we can check from a user perspective that this is working.

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
also look thru the referenced issue for any other things that we can test with this. it is quite trick to make sure this is working propertly.

File: pandas/tests/computation/test_eval.py (hunk 1 above)
Comment (saehuihwang) [reply pandas/tests/computation/test_eval.py:1868]: added

File: pandas/tests/computation/test_eval.py (hunk 2 above)
Comment (saehuihwang) [reply pandas/tests/computation/test_eval.py]: added

File: pandas/tests/computation/test_eval.py [hunk 3]
```diff
@@ -1865,6 +1865,32 @@ def test_invalid_engine():
...
+def test_numexpr_option_respected(use_numexpr):
+    # GH 32556
+    from pandas.core.computation.eval import _check_engine
+
+    with pd.option_context("compute.use_numexpr", use_numexpr):
+        result = _check_engine(None)
+        if use_numexpr:
+            assert result == "numexpr"
+        else:
+            assert result == "python"
```
Comment (MarcoGorelli) [inline pandas/tests/computation/test_eval.py:1879]: logic (e.g. if/else statements) in tests aren't ideal (https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html), how about something like
```
@pytest.mark.parametrize(
    ("use_numexpr", "expected"),
    (
          (True, "numexpr"),
          (False, "python"),
    )
)
def test_numexpr_option_respected(use_numexpr, expected):
```
and then, in the body, just `assert result == expected`?

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: pandas/tests/computation/test_eval.py (hunk 3 above)
Comment (saehuihwang) [reply pandas/tests/computation/test_eval.py]: This is good to know! Thank you for linking the resource. Added!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. over to you @MarcoGorelli

**Comment by jreback** [comment]
thanks @saehuihwang

**Pull request merged**